ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/metrics/ConsumptionSpeedMetricSpec.scala

```diff
@@ -0,0 +1,189 @@
+        recordTimeFunction = testRecordTimeFunction,
+      )
+
+      val (newMetric, periodicValue) = metric
+        .onNext(elem1)
+        .onNext(elem2)
+        .periodicValue()
+        ._1
+        .periodicValue()
+      val finalValue = newMetric.finalValue(totalDurationSeconds)
+
+      periodicValue shouldBe ConsumptionSpeedMetric.Value(Some(0.0))
+      finalValue shouldBe ConsumptionSpeedMetric.Value(None)
+    }
+
+    "correctly handle multiple periods with elements" in {
+      val totalDurationSeconds: Double = aPositiveDouble()
+      val elem1: String = aString()
+      val elem2: String = aString()
+      val elem3: String = aString()
+      val testNow = Clock.systemUTC().instant()
+      val recordTimes1 = aRecordTimesList(testNow)
+      // The assumption made here is that each consecutive element has higher record times
+      val recordTimes2 = aRecordTimesList(recordTimes1.last)
+      val recordTimes3 = aRecordTimesList(recordTimes2.last)
+      val periodMillis = aPositiveLong()
+      def testRecordTimeFunction: String => List[Timestamp] = recordTimeFunctionFromMap(
+        Map(
+          elem1 -> recordTimes1,
+          elem2 -> recordTimes2,
+          elem3 -> recordTimes3,
+        )
+      )
+
+      val metric = ConsumptionSpeedMetric.empty[String](
+        periodMillis = periodMillis,
+        recordTimeFunction = testRecordTimeFunction,
+      )
+
+      val (newMetric, periodicValue) = metric
+        .onNext(elem1)
+        .onNext(elem2)
+        .periodicValue()
+        ._1
+        .periodicValue()
+        ._1
+        .onNext(elem3)
+        .periodicValue()
+      val finalValue = newMetric.finalValue(totalDurationSeconds)
+
+      val firstElementOfThePeriod = recordTimes3.head
+      val lastElementOfThePeriod = recordTimes3.last
+      val expectedSpeed =
+        (lastElementOfThePeriod.getEpochSecond - firstElementOfThePeriod.getEpochSecond) * 1000.0 / periodMillis
+
+      periodicValue shouldBe ConsumptionSpeedMetric.Value(Some(expectedSpeed))
+      finalValue shouldBe ConsumptionSpeedMetric.Value(None)
+    }
+  }
+
+  private def aRecordTimesList(beforeInstant: Instant): List[Instant] =
+    (2 to 2 + Random.nextInt(10)).toList.map { _ =>
+      beforeInstant.minusSeconds(aPositiveLong())
+    }.sorted
+
+  private def recordTimeFunctionFromMap(
+      map: Map[String, List[Instant]]
+  )(str: String): List[Timestamp] =
+    map
+      .map { case (k, v) => k -> v.map(instantToTimestamp) }
+      .getOrElse(str, throw new RuntimeException(s"Unexpected record function argument: $str"))
+
+  private def instantToTimestamp(instant: Instant): Timestamp =
+    Timestamp.of(instant.getEpochSecond, instant.getNano)
+
+  private def dummyRecordTimesFunction(str: String): List[Timestamp] =
+    str.map(_ => Timestamp.of(aPositiveLong(), 0)).toList
+
+  private def aString(): String = Random.nextString(Random.nextInt(50))
+  private def aPositiveInt(): Int = Random.nextInt(100000)
```

Comment:
I do not see the benefit of randomizing the values on which the test operates. Randomness invites flakiness.

Reply:
My intention was to test more by property than by value. I would say that if we define properties clearly, then randomness is not a problem.
I could use property based checking framework to have these tests repeated and minimise the risk of flakiness or just hardcode some values. Which option would you prefer?

Reply:
Removed random number generating methods.